resize-observer/devicepixel2.html

```diff
@@ -0,0 +1,163 @@
+
+  const imageData = new ImageData(patternSize, patternSize);
+  imageData.data.set([
+    b, t, t, r,
+    t, b, g, t,
+    t, r, b, t,
+    g, t, t, b,
+  ].flat());
+  ctx.putImageData(imageData, 0, 0);
+  return ctx.canvas.toDataURL();
+}
+
+const patternURL = createPatternDataURL();
+
+/**
+ * Set the pattern's size on this element so that it draws were
+ * 1 pixel in the pattern maps to 1 devicePixel and then set
+ * its position so it's aligned to the pattern of the body element.
+ * The pattern is 'patternSize' large. x and y are the computed 
+ * devicePixel position of the element. To make the pattern align
+ * we need to set its position to -(x % patternSize), -(y % patternSize)
+ * in devicePixels
+*/
+function setPattern(elem, x, y) {
+  const oneDevicePixel = 1 / devicePixelRatio;
+  const patternPixels = oneDevicePixel * patternSize;
+  const xOff = -(x % patternSize);
+  const yOff = -(y % patternSize);
+  elem.style.backgroundImage = `url("${patternURL}")`;
+  elem.style.backgroundSize = `${patternPixels}px ${patternPixels}px`;
+  elem.style.backgroundPosition = `${xOff * oneDevicePixel}px ${yOff * oneDevicePixel}px`;
+}
+
+/*
+This test creates elements like this
+
+  <body>
+    <div id="front">
+      <div class="outer"><div></div><div></div></div>
+      <div class="outer"><div></div><div></div><div></div></div>
+      <div class="outer"><div></div><div></div><div></div><div></div></div>
+      ...
+    </div>
+  </body>
+
+It has the #front starting at the top left corner of the page, the
+same as the body so they should both have a devicePixel position of 0,0
+
+It then sets a repeating background-image and background-size on the the
+body element and all of the .outer children elements so they have a 4x4
+checkerboard image at a size that should draw each pixel of the 4x4 
+checkerboard pattern in single devicePixels.
+
+The devicePixelContentBoxSize of all the elements is queried with
+a ResizeObserver.
+
+Finally the background-position of each child element is set so that
+the patterns should align.
+
+In other words
+
+  <div class"outer">
+    <div></div><div></div><div></div>
+  </div>
+
+The first child will be displayed at left = 0 so its background-position is
+set to 0px 0px. The second child should be displayed at 
+
+secondChildLeft = entry(firstChild).devicePixelContentBoxSize[0].inlineSize. 
+
+The 3rd child should be displayed at 
+
+thirdChildLeft = entry(firstChild).devicePixelContentBoxSize[0].inlineSize + 
+                 entry(secondChild).devicePixelContentBoxSize[0].inlineSize
+
+If the element's left edge mod 4 is an non-zero value then we need to set the
+background-position of the element's pattern left edge mod 4 devicePixels.
+
+If what devicePixelContentBox reports is correct the children's patterns
+will align with the pattern in the body and the children will be invisible.
+If what devicePixelContentBox reports is incorrect the patterns will not
+align and it should be clearly visible.
```

Comment:
This comment needs a rewrite, if you take my above suggestion.

Reply:
done